src/models/Collectible.js

```diff
@@ -21,6 +21,12 @@
 // Types
 import type { Chain, ChainRecord } from 'models/Chain';
 
+/** Colectible is uniquly identified by both address & id */
+export type CollectibleId = {|
```

Comment:
Not sure what, but I have mixed feelings about this. Nothing major tho, but for me `id` sounds like single value identifier as either number or string. Checked throughout the code and I think there are 3 options to improve this:
1. Create a method which we can reuse in other screens/components/flows too called `getCollectibleId` and construct single string value `ID` using `tokenId` and `contractAddress` of `Collectible`.
2. Rename `CollectibleId` to `CollectibleContractMeta` or similar, but without `id` or `key`?
3. Just pass `Collectible` everywhere instead of `CollectibleId` and get needed values.

Reply:
It's a tricky subject. So according to ERC721 spec, an NFT is uniquely identified by (contract) address and (token) id. Hence I created this type, which I guess is quite useful in general, and I believe is more natural to use this form than string concat of address & id.

Naming is hard here, as we want a type to uniquely identify NFT, but part of that also called `id`. So maybe we can rename type to `CollectibleUniqueIdentifier`/`CollectibleUniqueId`?

Re option 2: the name is more confusing as contract meta seems very general term, and without the `id` part we are no longer uniquely identifying an NFT. Which defeats the purpose.

Re option 3: passing collectible around is an option, but imo that kinda repeating yourself, as you should have 1 source of truth for collectible defintion and then you should be able to reference the collectible without need to duplicate the whole thing. (compare to having asset definition in `supportedAsset` and referencing it with using token `address`)

Reply:
Yes, what makes NFTs unique is contract address and token ID.

I do agree this one is tricky, however I don't feel like putting both address and token ID into object and instead just use single value from `getCollectibleId` (option 1), wdyt?

Re option 2 I do agree that meta makes think of that there's more which leads to just option 3 which is the same as passing current object, but instead we don't add extra type to out codebase which you're aware has waaaaaay to many types that describe same thing which is another concern I have about this new `CollectibleId ` type.

Reply:
How about following solution:


```
const buildCollectibleKey = (contractAddress: string, tokenId: string) => `${valueAsKey(contractAddress)}-${tokenId}`
```

and additionally

```
const  getCollectibleKey = (collectible: Collectible) => buildCollectibleKey(collectible.contractAddress, collectible.id)
```